### ISSUE TITLE
[BUG] Can't compile in PlatformIO + VSCode

### ISSUE THREAD

**Issue by klcjr89** [MarlinFirmware/Marlin#14887]
I was able to compile the recent commits from last night this morning, but now when I try to do the same I get the following error: Not sure how to fix it honestly.

```
> Executing task: platformio run <

The terminal shell path "platformio" does not exist

Terminal will be reused by tasks, press any key to close it.
```

**Comment by kirilldobr** [comment]
Same thing

**Comment by kirilldobr** [comment]
Looks like the issue is with the PlatformIO itself.

**Comment by hcklawrence** [comment]
have the same problem just now, with no reason. I was doing arduino thing. it worked for a long time but suddenly comes with this error. 

after some google, this maybe the reason. 
http://docs.platformio.org/en/latest/faq.html#multiple-pio-cores-in-a-system 5

reinstall platformIO ide in vscode is not working for me. what i do is to install platformIO using pip, and it works as normal now. hope this help.

**Comment by kirilldobr** [comment]
Yep, it helped. Pretty weird though.

**Comment by hcklawrence** [comment]
> Yep, it helped. Pretty weird though.

I found a problem with this solution, I cannot use esp OTA to upload code now and it is so much slower in the OTA process...

**Comment by RudyBenner** [comment]
delete

.pioenvs
.piolib
.piolibdeps

directories, allow it to rebuild. could take a while.

works here. New directory structure will look like this

![image](https://user-images.githubusercontent.com/40361032/62748893-54762d80-ba28-11e9-862b-e8a5d7ee6997.png)

**Comment by hcklawrence** [comment]
> delete
> .pioenvs
> .piolib
> .piolibdeps
> directors, allow it to rebuild. could take a while.
> works here. New directory structure will look like this
> 
> ![image](https://user-images.githubusercontent.com/40361032/62744189-74e7bd00-ba13-11e9-92a8-6631783773e2.png)

this doesnt work in my situation. the platformio terminal has pio and platformio command. but the build and upload button (cmd + alt + b/u) are not working..

**Comment by klcjr89** [comment]
Installing via pip also worked for me but not sure why that was the solution.

**Comment by hcklawrence** [comment]
> > delete
> > .pioenvs
> > .piolib
> > .piolibdeps
> > directors, allow it to rebuild. could take a while.
> > works here. New directory structure will look like this
> > ![image](https://user-images.githubusercontent.com/40361032/62744189-74e7bd00-ba13-11e9-92a8-6631783773e2.png)
> 
> this doesnt work in my situation. the platformio terminal has pio and platformio command. but the build and upload button (cmd + alt + b/u) are not working..

I finally solved the problem by removing "~/.platformio", vs code will build a new symlink afterward, and it works. No idea what is the problem. 

and about the esp ota. I updated the platformio esp32 lib to 1.9 which need more detail infomation for the connection, for example like port for ota and so on, which is different with the esp32 manufactor so it was not working if not correctly stated. it is now fixed.

**Comment by codincafe** [comment]
> have the same problem just now, with no reason. I was doing arduino thing. it worked for a long time but suddenly comes with this error.
> 
> after some google, this maybe the reason.
> http://docs.platformio.org/en/latest/faq.html#multiple-pio-cores-in-a-system 5
> 
> reinstall platformIO ide in vscode is not working for me. what i do is to install platformIO using pip, and it works as normal now. hope this help.

This worked for me. I guess VSCode has been updated which is messing up the workspace

**Comment by tuxBurner** [comment]
I tried deleting ~/.platformio and ~/.vscode and the . directories in my project but still getting:

The terminal shell path "platformio" does not exist

**Comment by hpsaturn** [comment]
I have the same issue after update my last version of vscode, but I tried too delete these directories but the issue persist, from console all stuff works fine

**Comment by RudyBenner** [comment]
Should look like this. 

![image](https://user-images.githubusercontent.com/40361032/62801781-75846000-bab4-11e9-8fb8-1347a3138ef3.png)

**Comment by RudyBenner** [comment]
VSCODE 1.3.7 works fine here.

**Comment by RudyBenner** [comment]
![image](https://user-images.githubusercontent.com/40361032/62801913-bed4af80-bab4-11e9-9156-085f6367cc62.png)

and 

![image](https://user-images.githubusercontent.com/40361032/62801948-d57b0680-bab4-11e9-9693-dc1f0a2623ea.png)

**Comment by tuxBurner** [comment]
Think it is related to: https://github.com/platformio/platformio-vscode-ide/issues/957

**Comment by ph1nt** [comment]
Problem is in VSCode 1.37.0 on MacOS.
Solution for me:
1. delete '~/.platformio'
2. reinstall 'pip install platformio'
3. open VSCode and wait to rebuild configuration

Works too, but is uncomfortable:
1. open terminal inside VSCode
2. type platformio commands

**Comment by joaodematejr** [comment]
+1 Mac Os Mojave 10.14.6

**Comment by elektrik1980** [comment]
There is also such a problem when working with Arduino. Recently, everything worked.

**Comment by jsiqueir** [comment]
pip install platformio on Mac 10.13.6 with VSCode 1.37.0 worked fine.
Thank you!

**Comment by elektrik1980** [comment]
> Problem is in VSCode 1.37.0 on MacOS.
> Solution for me:
> 
>     1. delete '~/.platformio'
> 
>     2. reinstall 'pip install platformio'
> 
>     3. open VSCode and wait to rebuild configuration
> 
> 
> Works too, but is uncomfortable:
> 
>     1. open terminal inside VSCode
> 
>     2. type platformio commands
yes it worked. Linux Mint

**Comment by pliski** [comment]
> Problem is in VSCode 1.37.0 on MacOS.
This solved the problem also on Ubuntu 19.04.

> Problem is in VSCode 1.37.0 on MacOS.
> Solution for me:
> 
> 1. delete '~/.platformio'
> 2. reinstall 'pip install platformio'
> 3. open VSCode and wait to rebuild configuration
> 
> Works too, but is uncomfortable:
> 
> 1. open terminal inside VSCode
> 2. type platformio commands

This solved the problem also on Ubuntu 19.04.

**Comment by joaodematejr** [comment]
@jsiqueir @pliski  
it worked, thank you very much

**Comment by syproduction** [comment]
> Problem is in VSCode 1.37.0 on MacOS.
> Solution for me:
> 
> 1. delete '~/.platformio'
> 2. reinstall 'pip install platformio'
> 3. open VSCode and wait to rebuild configuration
> 
> Works too, but is uncomfortable:
> 
> 1. open terminal inside VSCode
> 2. type platformio commands

Worked on Mac OS 10.14.5
PIP didnt want to work, had to use SUDO

**Comment by p3p** [comment]
Resolved with VSCode PlatformIO IDE extension 1.8.2

**Comment by sl1pkn07** [comment]
if help, platformio vscode IDE have a option for use system instead of builtin piocore (downloaded throught platformio IDE)

in my macos, platformio is installed by homebrew, in my linux, with the package manager (archlinux/AUR), and in my windows, with PIP
